### PR TITLE
Add various functionality required to port the `script` program.

### PR DESCRIPTION
**Pull request by j-piecuch**
- Add `bswap{16,32,64}()` (implemented as macros using `__builtin_bswap{16,32,64}()`
- Add `login_tty()` from NetBSD and implement `openpty()` in `libutil`
- Implement `TIOCSCTTY` `ioctl()`

### CODE REVIEW

File: sys/kern/pty.c [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+  SCOPED_MTX_LOCK(&tty->t_lock);
+
+  while (uio->uio_resid > 0) {
+    if ((error = uiomove(&c, 1, uio)))
+      break;
+    if ((error = pty_putc_sleep(tty, pty, c))) {
+      if (error == ENOTTY)
+        error = 0;
+      break;
+    }
```
Comment (pj1031999) [inline sys/kern/pty.c:109]: In #910 that part of code looks different. Could you explain why?

**Review by pj1031999 — APPROVED** [review]
LGTM but please answer my questions.

File: include/sys/proc.h [hunk 2]
```diff
@@ -13,7 +13,6 @@
 
 typedef struct thread thread_t;
 typedef struct proc proc_t;
-typedef struct cred cred_t;
```
[the remark this answers is not in the document]
Comment (j-piecuch) [reply include/sys/proc.h]: Because `sys/cred.h` already has it. It's an unnecessary change, so I have reverted it.

File: sys/kern/pty.c (hunk 1 above)
Comment (j-piecuch) [reply sys/kern/pty.c]: I forgot to merge the recent changes to #910. My bad.

**Review by franciscozdo — APPROVED** [review]
LGTM

File: include/util.h [hunk 3]
```diff
@@ -2,6 +2,7 @@
 #define _UTIL_H_
 
 #include <sys/types.h>
+#include <sys/termios.h>
```
Comment (cahirwpz) [inline include/util.h:5]: The original header file performs forward declaration and does not include `termios.h`. I think it's better not to pollute visible symbol space.

**Review by cahirwpz — APPROVED** [review]
LGTM

**Pull request merged**